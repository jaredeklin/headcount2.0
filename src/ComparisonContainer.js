import React from 'react';
import Card from './Card';
import PropTypes from 'prop-types';
import './styles/comparisonContainer.css'

const ComparisonContainer = 
  ({selectedCards, comparison, selectCard, deselectCard}) => {
    let newDistricts = [];
    let newDistrictValues = [];
    if (comparison) {
      newDistricts = Object.keys(comparison);
      newDistrictValues = Object.values(comparison);
    }
    const comparisonCards = selectedCards.map((district) => {
      return (
        <Card location={district.location}
          stats={district.stats}
          key={district.location}
          selectCard={selectCard}
          deselectCard={deselectCard}
          className="card clicked"
        />);
    });
    
    return (
      <div className="comparison-container">
        {comparisonCards[0]}
        {selectedCards.length === 2 &&
        <div className="comparison-card">
          <h2>{newDistricts[0]}</h2>
          <p>{parseInt(newDistrictValues[0] * 100)}%</p>
          <h2>Comparision</h2>
          <p>{parseInt(newDistrictValues[2] * 100)}%</p>          
          <h2>{newDistricts[1]}</h2>
          <p>{parseInt(newDistrictValues[1] * 100)}%</p>
        </div>
      }
      {comparisonCards[1]}
    </div>
  )
}

ComparisonContainer.propTypes = {
  comparison: PropTypes.obj,
  selectCard: PropTypes.func,
  deselectCard: PropTypes.func,
  selectedCards: PropTypes.array
}


export default ComparisonContainer 