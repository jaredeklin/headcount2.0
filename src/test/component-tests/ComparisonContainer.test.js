import React from 'react';
import { shallow } from 'enzyme';
import ComparisonContainer from '../../ComparisonContainer';

describe('ComparisonContainer', () => {

  let mockData = [
    { statType : "Percent",
      stats: { 
        2004: 0.174,
        2005 : 0.202  
      } },
    { statType : "Percent",
      stats: { 
        2004: 0.474,
        2005 : 0.602  
      } }];
  let mockComparison = {
    'ACADEMY 20': 0.407,
    'ADAMS-ARAPAHOE 28J': 0.657,
    compared: 1.614
  };


  it('should match the snapshot', () => {
    let wrapper = 
      shallow(<ComparisonContainer 
        selectedCards={mockData} 
        comparison={mockComparison}
      />);
    
    expect(wrapper).toMatchSnapshot();
  });

  it('should turn decimals into percentages', () => {
    let wrapper = 
      shallow(<ComparisonContainer 
        selectedCards={mockData} 
        comparison={mockComparison}
      />);
    
    expect(wrapper.find('.district-percent-left').text()).toEqual('40%');
  });

  it('should not turn numbers into percentages', () => {
    let mockData = [
      { statType : "Number",
        stats: { 
          2004: 174,
          2005 : 202  } },
      { statType : "Number",
        stats: { 
          2004: 474,
          2005 : 602  
        } }];
    let mockComparison = {
      'ACADEMY 20': 136.333,
      "ADAMS COUNTY 14": 0,
      compared: null
    };
    let wrapper = 
      shallow(<ComparisonContainer 
        selectedCards={mockData} 
        comparison={mockComparison}
      />);
    
    expect(wrapper.find('.district-percent-left').text()).toEqual('136');
  });

  it('should do a comparison if stat is a number', () => {
    let wrapper = 
      shallow(<ComparisonContainer 
        selectedCards={mockData} 
        comparison={mockComparison}
      />);
    
    expect(wrapper.find('.district-comparison').text()).toEqual('161%');
  });

  it('should not do a comparison if stat is not a number', () => {
    let mockComparison = {
      'ACADEMY 20': 0.407,
      'ADAMS-ARAPAHOE 28J': 0.657,
      compared: Infinity
    };
    let wrapper = 
      shallow(<ComparisonContainer 
        selectedCards={mockData} 
        comparison={mockComparison}
      />);
    
    expect(wrapper.find('.district-comparison').text())
      .toEqual('not comparable to');
  });


});