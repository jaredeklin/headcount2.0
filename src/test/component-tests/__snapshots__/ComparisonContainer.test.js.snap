// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComparisonContainer should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ComparisonContainer
    comparison={
        Object {
            "ACADEMY 20": 0.407,
            "ADAMS-ARAPAHOE 28J": 0.657,
            "compared": 1.614,
          }
    }
    selectedCards={
        Array [
            Object {
              "statType": "Percent",
              "stats": Object {
                "2004": 0.174,
                "2005": 0.202,
              },
            },
            Object {
              "statType": "Percent",
              "stats": Object {
                "2004": 0.474,
                "2005": 0.602,
              },
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          className="card clicked"
          location={undefined}
          selectCard={undefined}
          statType="Percent"
          stats={
                    Object {
                              "2004": 0.174,
                              "2005": 0.202,
                            }
          }
/>,
        <div
          className="comparison-card"
>
          <header
                    className="header-left"
          >
                    <h4
                              className="district-name-left"
                    >
                              ACADEMY 20
                    </h4>
          </header>
          <header
                    className="header-right"
          >
                    <h4
                              className="district-name-right"
                    >
                              ADAMS-ARAPAHOE 28J
                    </h4>
          </header>
          <p
                    className="district-percent-left"
          >
                    41%
          </p>
          <p
                    className="district-percent-right"
          >
                    66%
          </p>
          <div
                    className="comparison-header"
          >
                    <p
                              className="district-percent-text"
                    >
                              *Average across all years
                    </p>
                    <h2
                              className="comparison-title"
                    >
                              Comparision
                    </h2>
          </div>
          <p
                    className="comparison-text"
          >
                    The average of 
                    ACADEMY 20
                     is
                    <span
                              className="district-comparison"
                    >
                              <span>
                                         
                                        161%
                                         of 
                              </span>
                    </span>
                    the average of  
                    ADAMS-ARAPAHOE 28J
                    .
          </p>
</div>,
        <Card
          className="card clicked"
          location={undefined}
          selectCard={undefined}
          statType="Percent"
          stats={
                    Object {
                              "2004": 0.474,
                              "2005": 0.602,
                            }
          }
/>,
      ],
      "className": "comparison-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": "card clicked",
          "location": undefined,
          "selectCard": undefined,
          "statType": "Percent",
          "stats": Object {
            "2004": 0.174,
            "2005": 0.202,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <header
              className="header-left"
>
              <h4
                            className="district-name-left"
              >
                            ACADEMY 20
              </h4>
</header>,
            <header
              className="header-right"
>
              <h4
                            className="district-name-right"
              >
                            ADAMS-ARAPAHOE 28J
              </h4>
</header>,
            <p
              className="district-percent-left"
>
              41%
</p>,
            <p
              className="district-percent-right"
>
              66%
</p>,
            <div
              className="comparison-header"
>
              <p
                            className="district-percent-text"
              >
                            *Average across all years
              </p>
              <h2
                            className="comparison-title"
              >
                            Comparision
              </h2>
</div>,
            <p
              className="comparison-text"
>
              The average of 
              ACADEMY 20
               is
              <span
                            className="district-comparison"
              >
                            <span>
                                           
                                          161%
                                           of 
                            </span>
              </span>
              the average of  
              ADAMS-ARAPAHOE 28J
              .
</p>,
          ],
          "className": "comparison-card",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <h4
                className="district-name-left"
>
                ACADEMY 20
</h4>,
              "className": "header-left",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "ACADEMY 20",
                "className": "district-name-left",
              },
              "ref": null,
              "rendered": "ACADEMY 20",
              "type": "h4",
            },
            "type": "header",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <h4
                className="district-name-right"
>
                ADAMS-ARAPAHOE 28J
</h4>,
              "className": "header-right",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "ADAMS-ARAPAHOE 28J",
                "className": "district-name-right",
              },
              "ref": null,
              "rendered": "ADAMS-ARAPAHOE 28J",
              "type": "h4",
            },
            "type": "header",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "41%",
              "className": "district-percent-left",
            },
            "ref": null,
            "rendered": "41%",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "66%",
              "className": "district-percent-right",
            },
            "ref": null,
            "rendered": "66%",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p
                  className="district-percent-text"
>
                  *Average across all years
</p>,
                <h2
                  className="comparison-title"
>
                  Comparision
</h2>,
              ],
              "className": "comparison-header",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "*Average across all years",
                  "className": "district-percent-text",
                },
                "ref": null,
                "rendered": "*Average across all years",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Comparision",
                  "className": "comparison-title",
                },
                "ref": null,
                "rendered": "Comparision",
                "type": "h2",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "The average of ",
                "ACADEMY 20",
                " is",
                <span
                  className="district-comparison"
>
                  <span>
                                     
                                    161%
                                     of 
                  </span>
</span>,
                "the average of  ",
                "ADAMS-ARAPAHOE 28J",
                ".",
              ],
              "className": "comparison-text",
            },
            "ref": null,
            "rendered": Array [
              "The average of ",
              "ACADEMY 20",
              " is",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <span>
                     
                    161%
                     of 
</span>,
                  "className": "district-comparison",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      " ",
                      "161%",
                      " of ",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    " ",
                    "161%",
                    " of ",
                  ],
                  "type": "span",
                },
                "type": "span",
              },
              "the average of  ",
              "ADAMS-ARAPAHOE 28J",
              ".",
            ],
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "className": "card clicked",
          "location": undefined,
          "selectCard": undefined,
          "statType": "Percent",
          "stats": Object {
            "2004": 0.474,
            "2005": 0.602,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            className="card clicked"
            location={undefined}
            selectCard={undefined}
            statType="Percent"
            stats={
                        Object {
                                    "2004": 0.174,
                                    "2005": 0.202,
                                  }
            }
/>,
          <div
            className="comparison-card"
>
            <header
                        className="header-left"
            >
                        <h4
                                    className="district-name-left"
                        >
                                    ACADEMY 20
                        </h4>
            </header>
            <header
                        className="header-right"
            >
                        <h4
                                    className="district-name-right"
                        >
                                    ADAMS-ARAPAHOE 28J
                        </h4>
            </header>
            <p
                        className="district-percent-left"
            >
                        41%
            </p>
            <p
                        className="district-percent-right"
            >
                        66%
            </p>
            <div
                        className="comparison-header"
            >
                        <p
                                    className="district-percent-text"
                        >
                                    *Average across all years
                        </p>
                        <h2
                                    className="comparison-title"
                        >
                                    Comparision
                        </h2>
            </div>
            <p
                        className="comparison-text"
            >
                        The average of 
                        ACADEMY 20
                         is
                        <span
                                    className="district-comparison"
                        >
                                    <span>
                                                 
                                                161%
                                                 of 
                                    </span>
                        </span>
                        the average of  
                        ADAMS-ARAPAHOE 28J
                        .
            </p>
</div>,
          <Card
            className="card clicked"
            location={undefined}
            selectCard={undefined}
            statType="Percent"
            stats={
                        Object {
                                    "2004": 0.474,
                                    "2005": 0.602,
                                  }
            }
/>,
        ],
        "className": "comparison-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "card clicked",
            "location": undefined,
            "selectCard": undefined,
            "statType": "Percent",
            "stats": Object {
              "2004": 0.174,
              "2005": 0.202,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <header
                className="header-left"
>
                <h4
                                className="district-name-left"
                >
                                ACADEMY 20
                </h4>
</header>,
              <header
                className="header-right"
>
                <h4
                                className="district-name-right"
                >
                                ADAMS-ARAPAHOE 28J
                </h4>
</header>,
              <p
                className="district-percent-left"
>
                41%
</p>,
              <p
                className="district-percent-right"
>
                66%
</p>,
              <div
                className="comparison-header"
>
                <p
                                className="district-percent-text"
                >
                                *Average across all years
                </p>
                <h2
                                className="comparison-title"
                >
                                Comparision
                </h2>
</div>,
              <p
                className="comparison-text"
>
                The average of 
                ACADEMY 20
                 is
                <span
                                className="district-comparison"
                >
                                <span>
                                                 
                                                161%
                                                 of 
                                </span>
                </span>
                the average of  
                ADAMS-ARAPAHOE 28J
                .
</p>,
            ],
            "className": "comparison-card",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <h4
                  className="district-name-left"
>
                  ACADEMY 20
</h4>,
                "className": "header-left",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "ACADEMY 20",
                  "className": "district-name-left",
                },
                "ref": null,
                "rendered": "ACADEMY 20",
                "type": "h4",
              },
              "type": "header",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <h4
                  className="district-name-right"
>
                  ADAMS-ARAPAHOE 28J
</h4>,
                "className": "header-right",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "ADAMS-ARAPAHOE 28J",
                  "className": "district-name-right",
                },
                "ref": null,
                "rendered": "ADAMS-ARAPAHOE 28J",
                "type": "h4",
              },
              "type": "header",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "41%",
                "className": "district-percent-left",
              },
              "ref": null,
              "rendered": "41%",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "66%",
                "className": "district-percent-right",
              },
              "ref": null,
              "rendered": "66%",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p
                    className="district-percent-text"
>
                    *Average across all years
</p>,
                  <h2
                    className="comparison-title"
>
                    Comparision
</h2>,
                ],
                "className": "comparison-header",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "*Average across all years",
                    "className": "district-percent-text",
                  },
                  "ref": null,
                  "rendered": "*Average across all years",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Comparision",
                    "className": "comparison-title",
                  },
                  "ref": null,
                  "rendered": "Comparision",
                  "type": "h2",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "The average of ",
                  "ACADEMY 20",
                  " is",
                  <span
                    className="district-comparison"
>
                    <span>
                                         
                                        161%
                                         of 
                    </span>
</span>,
                  "the average of  ",
                  "ADAMS-ARAPAHOE 28J",
                  ".",
                ],
                "className": "comparison-text",
              },
              "ref": null,
              "rendered": Array [
                "The average of ",
                "ACADEMY 20",
                " is",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <span>
                       
                      161%
                       of 
</span>,
                    "className": "district-comparison",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        " ",
                        "161%",
                        " of ",
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      " ",
                      "161%",
                      " of ",
                    ],
                    "type": "span",
                  },
                  "type": "span",
                },
                "the average of  ",
                "ADAMS-ARAPAHOE 28J",
                ".",
              ],
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "card clicked",
            "location": undefined,
            "selectCard": undefined,
            "statType": "Percent",
            "stats": Object {
              "2004": 0.474,
              "2005": 0.602,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
